### DISCUSSION TITLE
1.18.2 Terrain And Movements

### DISCUSSION THREAD

**Discussion by CommandCracker8** [ORelio/Minecraft-Console-Client#2008] [Q&A] [answered]
Hi. Does anyone know when Terrain and Movements will work for 1.18.2?

Thanks,

Commandcracker8

**Comment by ORelio** [comment]
We are lacking contributors / maintainers so unfortunately no ETA is available.

**Comment by Noksiowski** [comment]
what about 1.12?

**Reply by ORelio** [reply]
Terrain and movements works up to MC 1.16.5 at the moment.

**Comment by Daenges** [comment]
If your server supports multiple versions (Minigame servers often support clients down to 1.8) you can set the MCC version [here](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/Resources/config/MinecraftClient.ini#L26) to something **equal to or below** `1.16.5`. (MCC always selects the latest protocol version, on default `auto`)
Then your client reports an older version and the server starts communicating in a `terrain and movement` supported protocol version. Unfortunately you need to have a plugin like [this](https://www.spigotmc.org/resources/protocolsupport.7201/) installed on the server.

**Comment by milutinke** [comment]
@CommandCracker8 [BruceChenQAQ](https://github.com/BruceChenQAQ) made some huge progress in [milutinke#2](https://github.com/milutinke/Minecraft-Console-Client/pull/2)
We're currently testing the terrain movement and inventory handling and polishing it.
It might get merged with my branch today, so you could clone by branch and use it, until ORelio merges my PR with this repo.
If you just need a build, I can compile it for you.

Here is one video of a test (it's a bit older code, should be better now): https://streamable.com/vznqia

**Comment by ORelio** [comment, marked as answer]
1.18.2 terrain is now implemented in #1943
